index.d.ts

```diff
@@ -1,11 +1,23 @@
+
 // Type definitions for mockingcase 1.3
 // Project: https://github.com/strdr4605/mockingcase
 // Definitions by: Dragoș Străinu https://github.com/strdr4605
 
 declare module 'mockingcase' {
+
+  interface Options {
+    random?: boolean
+  }
+
   export = mockingcase;
-  function mockingcase(input: string | string[], options?: { random?: boolean }): string;
+  function mockingcase(input: string | string[], options?: Options): string;
+
   namespace mockingcase {
-    export function log(input: string | string[], options?: { random?: boolean }): void;
+    type MockingCase = (input: string, options?: Options) => string & {
```

Comment:
Can you please explain what does `string & {` means or share a link with explanations. Thank you!

Reply:
Full code block:
```javascript
type MockingCase = (input: string, options?: Options) => string & {
      log: (input: string, options?: Options) => void,
      overrideString: () => MockingCase
}
```

This section is for the `mockingcase` function:
`(input: string, options?: Options) => string`

Then we wrap `log` and `overrideString` in curly braces to list them within the type `MockingCase`.